lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.spec.ts

```diff
@@ -412,14 +433,190 @@ describe('ContentMetadataComponent', () => {
+            component.readOnly = false;
+            component.node.allowableOperations = null;
+            fixture.detectChanges();
+
+            expect(getToggleEditButton()).toBeNull();
+        });
+
+        it('should hide the tags edit button if node does not have `update` permissions', () => {
+            component.readOnly = false;
+            component.node.allowableOperations = null;
+            fixture.detectChanges();
+
+            expect(getTagsToggleEditButton()).toBeNull();
+        });
+
+        it('should hide the categories edit button if node does not have `update` permissions', () => {
+            component.readOnly = false;
+            component.node.allowableOperations = null;
+            fixture.detectChanges();
+
+            expect(getCategoriesToggleEditButton()).toBeNull();
+        });
+
+        it('should hide the groups edit button if node does not have `update` permissions', () => {
+            component.readOnly = false;
+            component.node.allowableOperations = null;
+            fixture.detectChanges();
+
+            expect(getGroupToggleEditButton()).toBeNull();
+        });
+    });
+
+    describe('hasToggleEdit', () => {
+        it('should return true when editable is false, readOnly is false, and hasAllowableOperations is true', () => {
+            component.editable = false;
+            component.readOnly = false;
+            component.hasAllowableOperations = true;
+            const result = component.hasToggleEdit;
+            expect(result).toBe(true);
+          });
+
+        it('should return false when editable is true', () => {
+            component.editable = true;
+            component.readOnly = false;
+            component.hasAllowableOperations = true;
+            const result = component.hasToggleEdit;
+            expect(result).toBe(false);
+        });
+    });
+
+    describe('hasTagsToggleEdit', () => {
+        it('should have hasTagsToggleEdit property as expected', () => {
+            component.editableTags = false;
+            component.readOnly = false;
+            component.hasAllowableOperations = true;
+            fixture.detectChanges();
+            expect(component.hasTagsToggleEdit).toBe(true);
+        });
+    });
+
+    describe('hasGroupToggleEdit', () => {
+        it('should return true when group is not editable, not read-only, and has allowable operations', () => {
+            component.readOnly = false;
+            component.hasAllowableOperations = true;
+            const group: CardViewGroup = {
+                title: 'Group Title',
+                properties: [],
+                expanded: true,
+                editable: false
+            };
+            const result = component.hasGroupToggleEdit(group);
+            expect(result).toBe(true);
+        });
+    });
+
+    describe('hasCategoriesToggleEdit', () => {
+        it('should have hasCategoriesToggleEdit property as expected', () => {
```

Comment:
Most of the tests added for this file are just checking component conditions and don't touch UI at all. Please check how the tests worked before, for example for tags&categories.

Reply:
@MichalKinas We have added the test cases according to the new changes and updated some of the test cases as per the new design changes.